plugins/field-grid-dropdown/src/index.ts

```diff
@@ -9,169 +9,215 @@
+      validator?: FieldGridDropdownValidator,
+      config?: FieldGridDropdownConfig) {
     super(menuGenerator, validator, config);
 
-    /**
-     * The number of columns in the dropdown grid. Must be an integer value
-     * greater than 0. Defaults to 3.
-     * @type {number}
-     * @private
-     */
-    this.columns_ = 3;
-    if (config && config['columns']) {
-      this.setColumnsInternal_(config['columns']);
+    if (config?.columns) {
+      this.setColumnsInternal_(config.columns);
     }
 
-    if (config && config['primaryColour']) {
-      this.primaryColour = config['primaryColour'];
+    if (config && config.primaryColour) {
+      this.primaryColour = config.primaryColour;
     }
 
-    if (config && config['borderColour']) {
-      this.borderColour = config['borderColour'];
+    if (config && config.borderColour) {
+      this.borderColour = config.borderColour;
     }
   }
 
   /**
    * Constructs a FieldGridDropdown from a JSON arg object.
-   * @param {!Object} options A JSON object with options.
-   * @returns {!FieldGridDropdown} The new field instance.
+   *
+   * @param config A JSON object with options.
+   * @returns The new field instance.
    * @package
    * @nocollapse
    */
-  static fromJson(options) {
-    // `this` might be a subclass of FieldGridDropdown if that class doesn't
+  static fromJson(config: FieldGridDropdownFromJsonConfig) {
+    if (!config.options) {
+      throw new Error(
+          'options are required for the dropdown field. The ' +
+          'options property must be assigned an array of ' +
+          '[humanReadableValue, languageNeutralValue] tuples.');
+    }
+    // `this` might be a subclass of FieldDropdown if that class doesn't
     // override the static fromJson method.
-    return new this(options['options'], undefined, options);
-  }
-
-  /**
-   * Sets the number of columns on the grid. Updates the styling to reflect.
-   * @param {number} columns The number of columns. Is rounded to
-   *    an integer value and must be greater than 0. Invalid
-   *    values are ignored.
-   * @private
-   */
-  setColumns(columns) {
-    this.setColumnsInternal_(columns);
-    this.updateColumnsStyling_();
+    return new FieldGridDropdown(config.options, undefined, config);
   }
 
+  /* eslint-disable @typescript-eslint/naming-convention */
   /**
    * Sets the number of columns on the grid.
-   * @param {?(string|number|undefined)} columns The number of columns. Is
-   *    rounded to an integer value and must be greater than 0. Invalid
-   *    values are ignored.
-   * @private
+   *
+   * @param columns The number of columns. Is rounded to an integer value and
+   *  must be greater than 0. Invalid values are ignored.
    */
-  setColumnsInternal_(columns) {
-    columns = parseInt(String(columns));
-    if (!isNaN(columns) && columns >= 1) {
-      this.columns_ = columns;
+  private setColumnsInternal_(columns: string | number) {
```

Comment:
The underscore can actually be removed here as well (since this method is private).

Here and for all other private methods (or it can be left for a follow-up)

Reply:
Done!